src/main/java/tictactoe/Grid.java

```diff
@@ -40,4 +40,15 @@ public String setSquare(int index, String data) {
     public int getSize() {
         return squares.size();
     }
+
+    public Grid copyGrid() {
+        var squares = copySquares();
+        return new Grid(squares);
+    }
+
+    private List<String> copySquares() {
+        List<String> copy = new ArrayList<>();
+        copy.addAll(squares);
```

Comment:
This works due to how Java handles strings. Can you think of any occasions where a copy made using this approach might result in a situation where editing the copy also edits the original? Why would such a thing happen?

Reply:
For example:

```
jshell> class MyClass {
   ...> private String myString;
   ...> public void setMyString(String str) { myString = str; }
   ...> public String getMyString() { return myString; }
   ...> }
|  created class MyClass

jshell> var a = new MyClass();
a ==> MyClass@306a30c7

jshell> a.setMyString("a");

jshell> var lst = new ArrayList<MyClass>();
lst ==> []

jshell> lst.add(a);
$5 ==> true

jshell> lst
lst ==> [MyClass@306a30c7]

jshell> var b = new ArrayList<MyClass>();
b ==> []

jshell> b.addAll(lst);
$8 ==> true

jshell> b
b ==> [MyClass@306a30c7]

jshell> var copy = b.get(0);
copy ==> MyClass@306a30c7

jshell> var original = lst.get(0);
original ==> MyClass@306a30c7

jshell> copy.setMyString("copy");

jshell> original.getMyString();
$13 ==> "copy"
```

Reply:
Thanks for the example, Ashley. I think if there was a `Square` class, and each square was an instance of it, using `copy.addAll(squares)` would add references to the existing squares to the new ArrayList. Would creating a new instance of each square and adding it to a new ArrayList be a suitable option?